typescript/src/util.ts

```diff
@@ -105,3 +105,16 @@ String.prototype.toSnakeCase = function(this: string): string {
   }
   return words.join('_');
 };
+
+String.prototype.CamelCaseBeforeDot = function(this: string): string{
+  const words = this.split('.');
+  const res:string[] = [];
+  for(var i = 0; i < words.length - 1; i++){
+    res.push(words[i].toCamelCase());
+  }
+  if(res.length > 0){
+    res.push(words[words.length - 1]);
+    return res.join('!.');
```

Comment:
This is TypeScript specifics in the code. We try not to do that.

Actually, I'm not even sure we need to have this function at all. Can we do it in the templates using a joiner? Dots are also language specific (we would've used `->` in C++ for example).

Reply:
this is for typescript type checking, that `request.crypto_key` always exists, so that `request.cropto_key.name` can be called.